### ISSUE TITLE
Fix false positives for selector lists and shared-line comments in comment-empty-line-before

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#2835]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where multiple shared-line comments using the `comment-empty-line-before` rule returns false results.

via https://github.com/stylelint/stylelint/tree/master/lib/rules/comment-empty-line-before
> _"If the comment is the very first node in a stylesheet then it is ignored. Shared-line comments are also ignored."_

> Which rule, if any, is this issue related to?

`comment-empty-line-before`

> What CSS is needed to reproduce this issue?

```css
/* comment */
.ok { /* comment */
  color: pink;
}

/* comment */
.foo, /* comment */
.bar { /* comment */
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

```js
{
  "rules": {
    'comment-empty-line-before': [ 'always', {
      ignore: ['stylelint-commands'],
    } ],
  }
}
```

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

n/a

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:
```shell
> test.css            
 8:8  ✖  Expected empty line before comment   comment-empty-line-before
```

**Comment by ntwb** [comment]
Here's a couple of failing tests that should pass:
```diff
diff --git lib/rules/comment-empty-line-before/__tests__/index.js lib/rules/comment-empty-line-before/__tests__/index.js
index be684390..3b867c25 100644
--- lib/rules/comment-empty-line-before/__tests__/index.js
+++ lib/rules/comment-empty-line-before/__tests__/index.js
@@ -76,6 +76,10 @@ testRule(
       {
         code: "a {\n\n  /* comment */\n  color: pink;\n}",
         description: "first-nested with empty line before"
+      },
+      {
+        code: "a {\n /* comment */\n  color: pink;\n}",
+        description: "shared-line comment should be ignored"
       }
     ],
 
@@ -147,6 +151,10 @@ testRule(
         code:
           "a {\ncolor: pink;\n/* stylelint-disable something */\ntop: 0;\n}",
         description: "no newline before a stylelint command comment"
+      },
+      {
+        code: "a {\n /* comment */\n  color: pink;\n}",
+        description: "shared-line comment should be ignored"
       }
     ]
   })
```

**Comment by jeddy3** [comment]
Will this be fixed by https://github.com/stylelint/stylelint/pull/2827?

Neither of the additional tests above contain _shared-line_ comments. So, I'm not sure those tests should pass.

**Comment by jeddy3** [comment]
I can replicate this with `8.1.1`. Very strange bug.

The following:

```css
any selector,
.bar { /* comment */
  color: pink;
}
```

Will produce a violation:

```
2:8 error Expected empty line before comment (comment-empty-line-before)
```

Whereas the following two produce no violations:

```css
.bar { /* comment */
  color: pink;
}
```

```css
any rule { }
.bar { /* comment */
  color: pink;
}
```

**Comment by hudochenkov** [comment]
[Demo with an issue](https://stylelint.io/demo#N4Igxg9gJgpiBcID0AqABJAtpmA7ALmikgDq4B0EA1msGqhhNnocWWowDYQBO8aABwCWuKgG4yAXzJkGWHASKkKAMwgQANPXTyWSsuQBGAQx61tjZora4Okbn0EjxUkBpABnAJ4FjADwRPMA8PN3AIXBUhAHNA4HY0EhAeAFdOGA8k-njbDkTwq3wAWhhMAXwvIs4RGCLDGDUeGCy0AG18404Ad2MvTLdaBLz8mNxeZoQ2pI8K9OqCIvljXCh+gF0htEk0DdzpXEkQSSA).
